scripts/lint-style.py

```diff
@@ -310,36 +303,39 @@ def format_errors(errors):
         if errno == ERR_WIN:
             output_message(path, line_nr, "ERR_WIN", "Windows line endings (\\r\\n) detected")
         if errno == ERR_TWS:
-            output_message(path, line_nr, "ERR_TWS", "Trailing whitespece detected on line")
+            output_message(path, line_nr, "ERR_TWS", "Trailing whitespace detected on line")
 
-def lint(path):
+def lint(path, fix=False):
     with path.open(encoding="utf-8", newline="") as f:
+        # We enumerate the lines so that we can report line numbers in the error messages correctly
+        # we will modify lines as we go, so we need to keep track of the original line numbers
         lines = f.readlines()
-        errs = line_endings_check(lines, path)
+        enum_lines = enumerate(lines, 1)
+        errs, newlines = line_endings_check(enum_lines, path)
         format_errors(errs)
-        lines = [line.rstrip() + "\n" for line in lines]
 
-        errs = long_lines_check(lines, path)
-        format_errors(errs)
-        errs = isolated_by_dot_semicolon_check(lines, path)
-        format_errors(errs)
-        (b, errs) = import_only_check(lines, path)
-        if b:
-            format_errors(errs)
-            return # checks below this line are not executed on files that only import other files.
-        errs = regular_check(lines, path)
-        format_errors(errs)
-        errs = small_alpha_vrachy_check(lines, path)
+        errs,newlines = long_lines_check(newlines, path)
         format_errors(errs)
-        errs = reserved_notation_check(lines, path)
+        errs,newlines = isolated_by_dot_semicolon_check(newlines, path)
         format_errors(errs)
-        errs = set_option_check(lines, path)
-        format_errors(errs)
-        errs = banned_import_check(lines, path)
+        errs,newlines = set_option_check(newlines, path)
         format_errors(errs)
+        if not import_only_check(newlines, path):
+            errs,newlines = regular_check(newlines, path)
+            format_errors(errs)
+            errs,newlines = banned_import_check(newlines, path)
+            format_errors(errs)
+    # if we haven't been asked to fix errors, or there are no errors or no fixes, we're done
+    if not fix or (not new_exceptions) or enum_lines == newlines:
+        return
+    path.with_name(path.name + '.bak').write_text("".join(l for _,l in newlines), encoding = "utf8")
+    shutil.move(path.with_name(path.name + '.bak'), path)
```

Comment:
Would it be better to move the existing file to `.bak`, to allow recovering it if this script does something wrong?